### DISCUSSION TITLE
How does `@collection.*` behave?

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#1840] [Q&A] [answered]
I want to implement a feature where I would provide invite keys to the users to sign up. To do so, I created another collection named `invitation_keys` with the following fields:
  > `key`: The key
  > `expire`: expiry date

In my `users` collection, I added the following create rule:
 ```
 (@request.data.key = @collection.invitation_keys.key) && (@collection.invitation_keys.expires > @now)
 ```
 
 Let's ignore expiry date verification for this question.
 
 ```
@request.data.key = @collection.invitation_keys.key
 ```
 My question is, will the above rule allow to create users if the request contain any key that exists in  the `invitation_keys` collection?

**Comment by ganigeorgiev** [comment, marked as answer]
> My question is, will the above rule allow to create users if the request contain any key that exists in the invitation_keys collection?

Not exactly. Since v0.11+ we have multi-match/match-all support which is the default for multi-valued expressions (including `@colleciton.*`) and the above rule will require every `invitation_keys.key` to match with the `@request.data.key`.

For an "at least one of" type of match you have to prefix the operator with `?` (_and also check for nonempty `@request.data.key` in case there are no invitation_keys_):

```js
@request.data.key != "" && @request.data.key ?= @collection.invitation_keys.key
```
